src/devices/Bno055/Bno055Sensor.cs

```diff
@@ -0,0 +1,495 @@
+
+namespace Iot.Device.Bno055
+{
+    public class Bno055Sensor : IDisposable
+    {
+        /// <summary>
+        /// The default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x28;
+
+        /// <summary>
+        /// This is the second I2C Address. It needs to be activated to be valid
+        /// </summary>
+        public const byte SecondI2cAddress = 0x29;
+
+        private const byte DeviceId = 0xA0;
+        private readonly bool _autoDispoable;
+        private readonly I2cDevice _i2cDevice;
+        private OperationMode _operationMode;
+        private Units _units;
+
+        /// <summary>
+        /// Get/Set the operation mode
+        /// </summary>
+        public OperationMode OperationMode
+        {
+            get { return _operationMode; }
+            set
+            {
+                _operationMode = value;
+                SetConfigMode(true);
+                SetOperationMode(_operationMode);
+                SetConfigMode(false);
+            }
+        }
+
+        /// <summary>
+        /// Set/Get the power mode
+        /// </summary>
+        public PowerMode PowerMode
+        {
+            get { return (PowerMode)ReadByte(Registers.PWR_MODE); }
+            set
+            {
+                SetConfigMode(true);
+                WriteReg(Registers.PWR_MODE, (byte)value);
+                SetConfigMode(false);
+            }
+        }
+
+        /// <summary>
+        /// Set/Get the temperature source
+        /// </summary>
+        public TemperatureSource TemperatureSource
+        {
+            get { return (TemperatureSource)ReadByte(Registers.TEMP_SOURCE); }
+            set
+            {
+                SetConfigMode(true);
+                WriteReg(Registers.TEMP_SOURCE, (byte)value);
+                SetConfigMode(false);
+            }
+        }
+
+        /// <summary>
+        /// Set/Get the units used. By default, international system is used
+        /// </summary>
+        public Units Units
+        {
+            get { return (Units)ReadByte(Registers.UNIT_SEL); }
+            set
+            {
+                SetConfigMode(true);
+                _units = value;
+                WriteReg(Registers.UNIT_SEL, (byte)_units);
+                SetConfigMode(false);
+            }
+        }
+
+        /// <summary>
+        /// Get the information about various sensor system versions and ID
+        /// </summary>
+        public Info Info { get; internal set; }
+
+        /// <summary>
+        /// Create an BNO055 sensor
+        /// </summary>
+        /// <param name="i2cDevice">The I2C Device</param>
+        /// <param name="operationMode">The operation mode to setup</param>
+        /// <param name="autoDispoable">true to dispose the I2C device at dispose</param>
+        public Bno055Sensor(I2cDevice i2cDevice, OperationMode operationMode = OperationMode.AccelerometerMagnetometerGyroscopeRelativeOrientation, bool autoDispoable = true)
```

Comment:
> AccelerometerMagnetometerGyroscopeRelativeOrientation

What a mouthful 😄

Reply:
You know, sensors where you can activate and deactivate some of the elements -:) And I know you guys prefer when it's explicit :-) Can't do longer :-)